code-challenges.js

```diff
@@ -2,31 +2,121 @@
 
 
 // --------------------1) Create a function that takes in an array of objects and returns a sentence about each person with their names capitalized.
-
 var people = [
   { name: "ford prefect", occupation: "hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "radio employee" }
 ]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
 
+// **My attempts below is to be able to destructor the object of "people" so that I can access the keys 'name', and 'occupation' so that I can call on them to use in a sentence.
+
+// let sentence = people.filter(key => key.name === 'arthur dent')
+
+// console.log(`${people.name} is a ${people.occupation}`);
+
+
+  // let sentence = people.filter(key => key.name && value => value.occupation)
+
+    // console.log(sentence(`${name} is a ${occupation}`));
+
+
+// let keys = Object.keys(people)
+// let values = Object.values(people)
+//
+//   let sentence = (keys, values) => {
+//     return `${this.keys} is a ${this.values}`
+//   }
+//
+//   console.log(sentence(`${this.keys} is a ${this.values}`));
+  // let entries = Object.entries(people)
+  //
+  //   let sentence = (entries) => {
+  //     return `${this.keys} is a ${this.values}`
+  //   }
+
+    // let sentence = sen.map(entries => {
+    //   return `${keys} is a ${values}`
+    // })
+    // let name = Object.keys(people)
+    // let occupation = Object.occupation(people)
+
+      let sentence = (people) => {
+        people.filter(key => key.name)
+        people.filter(job => job.occupation)
+        return `${key} is a ${job}`
+      }
+
+      console.log(sentence);
+
+// let sentence = (keys, values) =>{
+//   return `${keys} is a ${values}`
+// }
+
+// console.log(keys, values);
+// console.log(entries);
+
+
+
+
+
+
+// let arr = []
+
+
+// var { name, occupation } = people =>{
+//   return `${name} ${occupation}`
+// }
+// console.log(des(people));
+
+
 
```

Comment:
I know we talked about this one in class. There is clearly no lack of effort here! 
This would be a great practice challenge for you to continue to explore.
It is easy to get into a rabbit hole when the problem has so much information. Take a step back and focus on input/output and data types.